src/main/kotlin/org/digma/intellij/plugin/ui/common/EnvironmentsPanel.kt

```diff
@@ -170,27 +170,26 @@ class EnvironmentsPanel(
     }
 
     private fun rebuild() {
-
-        if (components.isNotEmpty()) {
-            this.components.forEach {
-                this.remove(it)
+        SwingUtilities.invokeLater {
+            if (components.isNotEmpty()) {
+                this.components.forEach {
+                    this.remove(it)
+                }
             }
```

Comment:
I think its better to move it with the same SwingUtilities.invokeLater before building the buttons under line 193